src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -514,6 +516,51 @@ static bool slash_boot_exists(void) {
         return (cache = false);
 }
 
+static bool fstab_is_same_device(const char *where, const char *what) {
+        _cleanup_endmntent_ FILE *f = NULL;
+        _cleanup_free_ char *dev = NULL;
+        struct mntent *me;
+        int r;
+
+        assert(where);
+        assert(what);
+
+        f = setmntent(fstab_path(), "re");
+        if (!f) {
+                assert(errno != ENOENT); /* We already know the path has an fstab entry, so fstab must exist */
+
+                log_debug_errno(errno, "Failed to open '%s', ignoring: %m", fstab_path());
+                return false;
+        }
+
+        while ((me = getmntent(f)))
+                if (path_equal(where, me->mnt_dir)) {
+                        _cleanup_free_ char *p = NULL;
+
+                        p = fstab_node_to_udev_node(me->mnt_fsname);
+                        if (!p) {
+                                log_oom_debug();
+                                return false;
+                        }
+
+                        free_and_replace(dev, p);
+                }
+
+        assert(dev);
+
+        if (!is_device_path(dev))
+                return false;
+
+        if (path_equal(dev, what))
+                return true;
+
+        r = devnode_same(dev, what);
+        if (r < 0)
+                log_debug_errno(r, "Failed to check if fstab entry for '%s' is the same device as '%s', assuming not: %m",
+                                where, what);
```

Comment:
Please propagate the error.

Reply:
udev might not be running by the time we call `devnode_same`, where `stat` would return `errno = ENOENT`. We should still try `/efi/` in that case.